pkg/controller/controller.go

```diff
@@ -726,7 +726,7 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 	pvName := req.Name
 	provisionerCredentials := req.Secrets
 
-	createCtx, cancel := context.WithTimeout(ctx, p.timeout)
+	createCtx, cancel := context.WithTimeout(markAsMigrated(ctx), p.timeout)
```

Comment:
I think one more thing you want to do is to redefine the metrics 
https://github.com/kubernetes-csi/external-provisioner/blob/aef465af4d2649eda1b8e63c9f8db9be257b27b5/cmd/csi-provisioner/csi-provisioner.go#L183

Here change the definition to 
```
metricsManager := metrics.NewCSIMetricsManagerWithOptions(
    "" /* driverName */,
    metrics.WithMigration(),
)
```